engine/job.go

```diff
@@ -69,16 +63,21 @@ func (job *Job) Run() error {
 	if job.Eng.Logging {
 		log.Infof("+job %s", job.CallString())
 		defer func() {
-			// what if err is nil?
-			log.Infof("-job %s%s", job.CallString(), job.err)
+			var okerr string
+			if err != nil {
+				okerr = fmt.Sprintf("ERR: %s", err)
+			} else {
+				okerr = "OK"
+			}
+			log.Infof("-job %s %s", job.CallString(), okerr)
 		}()
 	}
 	var errorMessage = bytes.NewBuffer(nil)
 	job.Stderr.Add(errorMessage)
 	if job.handler == nil {
-		job.err = fmt.Errorf("%s: command not found", job.Name)
+		err = fmt.Errorf("%s: command not found", job.Name)
 	} else {
-		job.err = job.handler(job)
+		err = job.handler(job)
 		job.end = time.Now()
 	}
 	if job.closeIO {
```

Comment:
Shouldn't we be making sure all the streams are at least attempt to be closed before returning anyway?